### DISCUSSION TITLE
glTF support in OpenSceneGraph

### DISCUSSION THREAD

**Discussion by LudwigFriedmann** [openscenegraph/OpenSceneGraph#1262] [General]
I'd like to bring up a topic that [has been around for a while](http://lists.openscenegraph.org/pipermail/osg-users-openscenegraph.org/2020-February/010831.html). An importer/exporter for [Khronos glTF](https://github.com/KhronosGroup/glTF) is already existant in [osgEarth](https://github.com/gwaldron/osgearth/tree/master/src/osgEarthDrivers/gltf). Wouldn't it make sense to have this implemented at a central place so that all OSG implementations can benefit from it? I think that glTF support would be a great gain for OSG...

**Comment by robertosfield** [comment]
The OpenSceneGraph project like OpenGL is now in maintenance mode so I'm not personally looking to add new features to the project - I simply don't have the time available.  The OSG's successor [VulkanSceneGraph](http://www.vulkanscenegraph.org) already had glTF support thanks to the Assimp in the vsgXchange library.

**Comment by LudwigFriedmann** [comment]
Hello Robert and please excuse the late reply. I haven't had any contact with VulkanSceneGraph yet, but I'll be happy to take a closer look at the project, especially with regard to the portability of existing OSG projects. Best regards, Ludwig

**Reply by robertosfield** [reply]
Unfortunately existing OSG applications aren't directly portable to the VSG, the VSG is wholly new scene graph project, while it shares much of the DNA (I'm the lead author of both) is a next gen scene graph, and will require applications scene graph parts to be substantially rewritten.

However, while a rewrite of scene graph parts will be required to port to the VSG, the VSG is the closest to the OSG you'll find in design philosophy and implementation.  Some areas you'll find the OSG still had has more features as it's a longer established project, but where the VSG does provide equivalent features they are a step beyond what the OSG is capable of doing.  

As time goes on are filling in the gaps in features.  For some applications the VSG may already be more feature rich - such as support for glTF, while others may find OSG features missing or not yet completed.  Where you application might fit along this spectrum I can not say.

**Comment by xarray** [comment]
FYI, osgverse now has a complete GLTF reader based on tinygltf. It supports loading gltf/glb/b3dm formats and parsing static meshes, textures, skeletons, character animations and blendshapes.
https://github.com/xarray/osgverse
